ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -465,6 +490,328 @@ const triggerSet: TriggerSet<Data> = {
+    },
+    {
+      id: 'P12S Engravement 3 Theos Drop AoE',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravement3TheosSoulIds },
+      condition: (data, matches) => data.engravementCounter === 3 && data.me === matches.target,
+      delaySeconds: (_data, matches) => parseFloat(matches.duration) - 4,
+      alertText: (_data, matches, output) => {
+        const mark = matches.effectId === engravementIdMap.crossMarked ? '+' : 'x';
+        return output.dropBait!({ mark: mark });
+      },
+      outputStrings: {
+        dropBait: {
+          en: 'Drop baited ${mark} AoE',
+        },
+      },
+    },
+    {
+      id: 'P12S Engravement 3 Theos Bait Adds',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravement3TheosSoulIds },
+      condition: (data, matches) => data.engravementCounter === 3 && data.me === matches.target,
+      delaySeconds: (_data, matches) => parseFloat(matches.duration),
+      alertText: (_data, _matches, output) => output.baitCleave!(),
+      outputStrings: {
+        baitCleave: {
+          en: 'Bait add line cleave',
+        },
+      },
+    },
+    {
+      id: 'P12S Engravement 3 Towers Collect',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravementTowerIds },
+      condition: (data) => data.engravementCounter === 3,
+      run: (data, matches) => {
+        data.engravement3TowerPlayers.push(matches.target);
+        data.engravement3TowerType = engravementLabelMap[matches.effectId];
+      },
+    },
+    {
+      id: 'P12S Engravement 3 Towers Initial',
+      type: 'GainsEffect',
+      netRegex: { effectId: engravementTowerIds, capture: false },
+      condition: (data) => data.engravementCounter === 3,
+      delaySeconds: 0.3,
+      suppressSeconds: 1,
+      response: (data, _matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          towerOnYou: {
+            en: '${color} Tower on You (w/ ${partner})',
+          },
+          towerLater: {
+            en: '${color} Towers later',
+          },
+          light: {
+            en: 'Light',
+          },
+          dark: {
+            en: 'Dark',
+          },
+          unknown: Outputs.unknown,
+        };
+        const towerOnYou = data.engravement3TowerPlayers.indexOf(data.me) !== -1;
+        let towerColor = output.unknown!();
+        if (data.engravement3TowerType !== undefined)
+          towerColor = data.engravement3TowerType === 'lightTower'
+            ? output.light!()
+            : output.dark!();
+        let partner;
+        if (towerOnYou) {
+          partner = data.ShortName(data.engravement3TowerPlayers.find((name) =>
+            name !== data.me
+          )) ?? output.unknown!();
+          return { alertText: output.towerOnYou!({ color: towerColor, partner: partner }) };
+        } else if (data.party.isDPS(data.me)) {
+          // DPS want to know tower color to know if they are soaking or baiting add cleaves later
```

Comment:
I'm always a little hesitant on the `data.party` checks, both because of https://github.com/OverlayPlugin/OverlayPlugin/issues/60 but also because of (far far) future `BLU` stuff.  Could you track who has the tethers, what color they are, and tell them more specifically what they're doing, e.g. bait cleaves later or bait middle dark tower later or bait non-middle(?) light tower later?